benches/benches/bevy_ecs/components/change_detection.rs

```diff
@@ -10,10 +10,25 @@ use rand_chacha::ChaCha8Rng;
 
 #[derive(Component, Default)]
 #[component(storage = "Table")]
-struct Table;
+struct Table(f32);
 #[derive(Component, Default)]
 #[component(storage = "SparseSet")]
-struct Sparse;
+struct Sparse(f32);
+
+trait BenchModify {
+    fn bench_modify(&mut self);
+}
+
+impl BenchModify for Table {
+    fn bench_modify(&mut self) {
+        self.0 += 1f32;
```

Comment:
This doesn't change the situation since this function returns `()`, also a ZST that is being `black_box`'ed. Might be better to just directly use this in the benchmarks:

```rust
self.0 += 1f32;
black_box(self.0)
```

Reply:
How would that work with the generic trait over Table and Sparse ? I could deref..? Or should we just keep the redundancy for now and address generics on a follow up PR ?

Reply:
Oh you probably mean that I change the implementation of each bench_modify ; I can do that tomorrow 👍